custom_components/ttlock/coordinator.py

```diff
@@ -139,14 +145,26 @@ async def _async_update_data(self) -> LockState:
                 try:
                     state = await self.api.get_lock_state(self.lock_id)
                     new_data.locked = state.locked == State.locked
+                    new_data.opened = state.opened == SensorState.opened
                 except Exception:
                     pass
 
             new_data.auto_lock_seconds = details.autoLockTime
+            new_data.auto_lock = True if details.autoLockTime > 0 else False
+            new_data.lock_sound = details.lockSound == 1
+
             new_data.passage_mode_config = await self.api.get_lock_passage_mode_config(
                 self.lock_id
             )
-
+            
+            if Features.door_sensor in Features.from_feature_value(details.featureValue):
+                sensor = await self.api.get_sensor(self.lock_id)
+                if sensor.battery_level is not None:
+                    new_data.sensor_battery = sensor.battery_level
+                    new_data.sensor = True
+                else:
+                    new_data.sensor = False
```

Comment:
Is this just for polling the battery level?
If so it might be worth considering storing a timestamp of the last attempted fetch alongside `sensor` and `sensor_battery` and gating the fetch on some interval? I suspect this is what you meant in the PR description about polling every 24h

Reply:
So this query is for sensor info only (name,Mac,bat) and does not give the state. So yes really only pulling the battery, The sensor bool was my way of filtering if a lock has a sensor (which is pointless but I couldn't figure out a better way)